migrations/sqls/20241116085102-initialize-up.sql

```diff
@@ -0,0 +1,275 @@
+-- back up with old schema
+alter schema badgehub rename to badgehub_old;
+create schema badgehub;
+create table badgehub.migrations
+(
+    like badgehub_old.migrations including all
+);
+
+-- create tables
+
+create table badges
+(
+    slug       text not null primary key,
+    name       text,
+    created_at timestamptz default now(), -- creation timestamp
+    updated_at timestamptz default now(), -- update timestamp
+    deleted_at timestamptz                -- soft delete timestamp (nullable)
+);
+
+create table users
+(
+    id                text primary key,          -- using text as recommended
+    email             text unique,               -- using text as recommended
+    admin             boolean,
+    name              text not null,
+    password          text not null,
+    remember_token    text,
+    editor            text,
+    public            boolean,
+    show_projects     boolean,
+    email_verified_at timestamptz,               -- email verification timestamp
+    created_at        timestamptz default now(), -- to capture creation timestamps
+    updated_at        timestamptz default now(), -- to track updates
+    deleted_at        timestamptz                -- soft delete timestamp (nullable)
+);
+
+create table projects
+(
+    created_at       timestamptz not null default now(),
+    updated_at       timestamptz not null default now(),
+    deleted_at       timestamptz,
+    version_id       integer,
+    user_id          text        not null,
+    slug             text        not null primary key,
+    git              text,
+    allow_team_fixes boolean,
+    constraint projects_user_id_fk foreign key (user_id) references users (id) on delete cascade -- if user is deleted, delete his projects
```

Comment:
`delete their projects`

Reply:
Yes true, if a user really get's deleted out of the DB as opposed to the deleted_at being set, then all the users' projects get deleted as well. 
This way we can keep a more clean DB with stronger constraints about the validity of the data.

Reply:
No I meant `their` as in, singular they (which is gender neutral in English)

Reply:
Oh sorry completely missed  that, let me update that.